src/ledger/LedgerState.h

```diff
@@ -14,6 +14,157 @@
+//                |    |+---------------------------------------------------+|
+//                |    ||LedgerState::Impl                                  ||
+//                |    ||                                                   ||
+//                +------AbstractLedgerStateParent &mParent                 ||
+//                     ||AbstractLedgerState *mChild = nullptr              ||
+//                     ||                                                   ||
+//  +----------------+ ||+------------------------------+ +----------------+||
+//  |LedgerStateEntry| |||mActive                       | |mEntry          |||
+//  |(for client use)| |||                              | |                |||
+//  |                | |||map<LedgerKey,                | |map<LedgerKey,  |||
+//  |weak_ptr<Impl>  | |||    shared_ptr<EntryImplBase>>| |    LedgerEntry>|||
+//  +----------------+ ||+------------------------------+ +----------------+||
+//           |         |+---------------------------------------------------+|
+//                     +-----------------------------------------------------+
+//           |                                          ^
+//                       +-------------------------+    |  +-------------+
+//           |           |+-------------------------+   |  |+-------------+
+//                       ||+-------------------------+  |  ||+-------------+
+//           |           |||LedgerStateEntry::Impl   |  |  +||LedgerEntry  |
+//         weak - - - - >|||(indicates "entry is     |  |   +|(XDR object) |
+//                       |||active in this state")   |  |    +-------------+
+//                       |||                         |  |           ^
+//                       +||AbstractLedgerState &  -----+           |
+//                        +|LedgerEntry &          -----------------+
+//                         +-------------------------+
+//
+//
+// The following notes may help with orientation and understanding:
+//
+//  - A LedgerState is an in-memory transaction-in-progress against the
+//    ledger in the database. Its ultimate purpose is to model a collection
+//    of LedgerEntry (XDR) objects to commit to the database.
+//
+//  - At any given time, a LedgerState may have zero-or-one active
+//    _sub_-LedgerStates (sub-transactions), arranged in a parent/child
+//    relationship. The terms "parent" and "child" refer exclusively to
+//    this nesting-relationship of transactions. The presence of an active
+//    sub-LedgerState is indicated by a non-null mChild pointer.
+//
+//  - Once a child is closed and the mChild pointer is reset to null,
+//    a new child may be opened. Attempting to open two children at once
+//    will throw an exception.
+//
+//  - The entries to be committed in each transaction are stored in the
+//    mEntry map, keyed by LedgerKey. This much is straightforward!
+//
+//  - Committing any LedgerState / transaction except the root will merge
```

Comment:
This note and the next are almost, but not exactly, correct. The distinction is that the root cannot be committed explicitly, it writes (and subsequently commits) to the underlying database automatically when its child is committed. So commit always merges the entries of the child into its parent's entries, which specifically means writing to the database if the parent is the root.

Reply:
Ah, ok. WIll push fix in a moment, thanks for clarifying!